exercises/hamming/test/Tests.hs

```diff
@@ -102,3 +110,65 @@ cases = [ Case { description = "empty strands"
                , expected    = Nothing
                }
         ]
+
+hammingGen :: Int -> Gen Case
+hammingGen 0 = return
+  Case { description = "Empty strands have no distance"
+       , strand1 = ""
+       , strand2 = ""
+       , expected = Just 0
+       }
+
+hammingGen n = do
+  (a, b, count) <- nucleotidePairGen
+  case' <- hammingGen (n-1)
+  return . hammingDesc $
+    case' { description = ""
+          , strand1 = a ++ strand1 case'
+          , strand2 = b ++ strand2 case'
+          , expected = (+) <$> count <*> expected case'
+          }
+
+hammingDesc :: Case -> Case
+hammingDesc case'@Case{..} = case' { description = desc' }
+  where
+    desc' = "Strands of " ++ length' ++ " with " ++ dist' ++ " distance"
+    dist' = maybe "invalid" show expected
+    length' = case comparing length strand1 strand2 of
+      EQ -> "lengths " ++ show (length strand1)
+      _  -> "different lengths " ++ show (length strand1) ++ " and " ++
```

Comment:
optional: I guess this is not yet at the point where I would start suggesting `printf`, but I was considering it

Reply:
There may be a neater way to express this.